manifests/repo/apt/cloudsmith.pp

```diff
@@ -0,0 +1,30 @@
+class erlang::repo::apt::cloudsmith (
+  String $ensure      = $erlang::repo_ensure,
```

Comment:
can you review the datatypes here? I think they can be more strict. $ensure for example could be an enum, and location/key_source could be https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/types/httpsurl.pp